alpha/template/semver/types.go

```diff
@@ -27,6 +27,7 @@ type semverTemplate struct {
 	Schema                string                       `json:"schema"`
 	GenerateMajorChannels bool                         `json:"generateMajorChannels,omitempty"`
 	GenerateMinorChannels bool                         `json:"generateMinorChannels,omitempty"`
+	ChannelTypePreference streamType                   `json:"channelTypePreference,omitempty"`
```

Comment:
+1 on the rename.  

Expectation for type-mismatch here is no-op, since the underlying implementation is to tie-break indifferentiable channels and selecting only one of the generate options would entirely miss that code. 

Adding new cases to capture/enforce.

Reply:
> selecting only one of the generate options would entirely miss that code.

Agreed, but it seems like that would be an obvious indication that the input was a user accident or misunderstanding. I think I'd suggest that we:
1. Fail if an unknown defaultChannelTypePreference value is given
2. Fail if the defaultChannelTypePreference doesn't make sense based on the generate options.

Reply:
> Fail if the defaultChannelTypePreference doesn't make sense based on the generate options.

This currently would be a surprise if the author was generating major channels, since the preference defaults to minor channels.  
I had preferred not to do this because it meant that `GenerateMajorChannels: true` required `DefaultChannelTypePreference: major`, but that's a result of the flag defaulting implementation.  To do this we have to shift around how we weight toward minor channel default tiebreak... so working on that.